packages/ui-responsive/src/addElementQueryMatchListener.ts

```diff
@@ -76,27 +80,35 @@ import { debounce } from '@instructure/debounce'
  * ```
  * @module addElementQueryMatchListener
  * @param {Object} query - object consisting of names and query objects
- * @param {ReactComponent|DomNode} el - a DOM node or a function returning a DOM node
+ * @param {Node|Window|React.ReactElement|React.Component|function} el - a DOM node or a function returning a DOM node
  * @param {function} cb - called with an array of the names of the currently
  * matching queries whenever a matching query changes
  * @returns {function} remove() function to call to remove the listener
  */
-// @ts-expect-error ts-migrate(7006) FIXME: Parameter 'query' implicitly has an 'any' type.
-function addElementQueryMatchListener(query, el, cb) {
+function addElementQueryMatchListener(
+  query: BreakpointQueries,
+  el:
+    | Node
+    | Window
+    | React.ReactElement
+    | React.Component
+    | ((
+        ...args: any[]
+      ) => Node | Window | React.ReactElement | React.Component),
+  cb: (queriesMatching: QueriesMatching) => any
```

Comment:
`cb` is `UpdateMatches` type (from index.tsx)

The signature of this method should be the same as `addMediaQueryMatchListener`, I'd extract it to a type.

Reply:
otherwise it looks good!

Reply:
`addMediaQueryMatchListener` sadly has an extra parameter `matchMedia` :S I couldn't find a way to extract it in a way so one can have an extra parameter and the other doesn't.